firestore/query.go

```diff
@@ -701,11 +701,11 @@ func newQueryDocumentIterator(ctx context.Context, q *Query, tid []byte) *queryD
 }
 
 func (it *queryDocumentIterator) next() (_ *DocumentSnapshot, err error) {
-	it.ctx = trace.StartSpan(it.ctx, "cloud.google.com/go/firestore.Query.RunQuery")
-	defer func() { trace.EndSpan(it.ctx, err) }()
-
 	client := it.q.c
 	if it.streamClient == nil {
+		it.ctx = trace.StartSpan(it.ctx, "cloud.google.com/go/firestore.Query.RunQuery")
```

Comment:
it does seem to me this would get us the first response, which I suppose is the closest thing to 'time to run query'

We could have two spans started and also end one on L729, when we detect the end of the stream.

@ahmetb any thoughts on what would be useful from your perspective?

Reply:
I am alright merging this in. I can open issues to track additional work :)